src/lem/mod.rs

```diff
@@ -155,6 +196,27 @@ impl LEMOP {
         Ok(())
     }
 
+    /// Recursively computes the number of slots needed for a LEMOP
+    pub fn num_hash_slots(&self) -> NumSlots {
+        match self {
+            LEMOP::Hash2(..) | LEMOP::Unhash2(..) => NumSlots::new((1, 0, 0)),
+            LEMOP::Hash3(..) | LEMOP::Unhash3(..) => NumSlots::new((0, 1, 0)),
+            LEMOP::Hash4(..) | LEMOP::Unhash4(..) => NumSlots::new((0, 0, 1)),
+            LEMOP::MatchTag(_, cases) => cases
+                .values()
+                .fold(NumSlots::default(), |acc, op| acc.max(&op.num_hash_slots())),
+            LEMOP::MatchSymPath(_, cases, def) => {
+                cases.values().fold(def.num_hash_slots(), |acc, op| {
+                    acc.max(&op.num_hash_slots())
+                })
+            }
+            LEMOP::Seq(ops) => ops
+                .iter()
+                .fold(NumSlots::default(), |acc, op| acc.add(&op.num_hash_slots())),
```

Comment:
Just a sanity check and reminder to myself: I think is not doing any hash deduplication yet.

Reply:
No yet. This will happen in next task.